src/core/TSDB.java

```diff
@@ -569,6 +572,8 @@ public WritableDataPoints newDataPoints() {
     } else {
       v = Bytes.fromLong(value);
     }
+    collector.record("datapoints.received", datapoints_received, "type=all");
```

Comment:
Don't collect this every time, just when the ``collectStats()`` method is called.